File: test-suite/src/test/java/io/micronaut/docs/context/annotation/primary/Blue.java
package io.micronaut.docs.context.annotation.primary;

import io.micronaut.context.annotation.Requires;

//tag::imports[]

import javax.inject.Singleton;

//end::imports[]

@Requires(property = "spec.name", value = "primaryspec")
//tag::clazz[]
@Singleton
public class Blue implements ColorPicker {

    @Override
    public String color() {
        return "blue";
    }
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/context/annotation/primary/Blue.kt
package io.micronaut.docs.context.annotation.primary

import io.micronaut.context.annotation.Requires
//tag::imports[]

import javax.inject.Singleton

//end::imports[]

@Requires(property = "spec.name", value = "primaryspec")
//tag::clazz[]
@Singleton
class Blue: ColorPicker {
    override fun color(): String {
        return "blue"
    }
}
//end::clazz[]
